### PR TITLE
[DataGridPremium] Bootstrap `@mui/x-data-grid-premium`

### LINKED ISSUE #4090: [data-grid] Setup a package to publish the DataGridPremium.
# Summary 
We'll publish our premium features with a third component on a third package, and so we need to establish the new structure and migrate the experimental features (Row grouping and Excel export) there.

# Objective
Essentially what we're going for is the following import structure:

`import { DataGridPremium } from '@mui/x-data-grid-premium';`

# Why?
It’s very clear, by reading the imports of a demo, which plan the users would need. 
And It's the fastest way to achieve:
* Clear imports
* Clear typing of exported interfaces
* Clear typing of props
* Tree shaking.

### PR DESCRIPTION
**Pull request by flaviendelangle**
DO NOT MERGE BEFORE PREMIUM PLAN IS READY

Closes #4090

[Working codesandbox](https://codesandbox.io/s/datagridpremium-v5-quick-start-forked-v701kh?file=/src/App.tsx)

- [x] Create a new package `@mui/x-data-grid-premium`
- [x] Move row grouping in `@mui/x-data-grid-premium` and drop its experimental feature
- [x] Add a warning when `props.experimentalFeature.rowGrouping: true` on `DataGridPro`
- [x] Move excel export in `@mui/x-data-grid-premium` and drop its experimental feature
- [x] Add a warning when `props.experimentalFeature.excelExport: true` on `DataGridPro`
- [x] Use `@mui/x-data-grid-premium` in `@mui/x-data-grid-generator`
- [x] Adapt the api building scripts to handle premium-only properties
- [ ] Check the plan of the license

image

### CODE REVIEW

File: packages/grid/x-data-grid-pro/src/DataGridPro/useDataGridProProps.ts [hunk 1]
```diff
@@ -21,13 +22,19 @@ export const DATA_GRID_PRO_PROPS_DEFAULT_VALUES: DataGridProPropsWithDefaultValu
...
   disableChildrenFiltering: false,
   disableChildrenSorting: false,
-  rowGroupingColumnMode: 'single',
   getDetailPanelHeight: () => 500,
 };
 
+const rowGroupingWarning = buildWarning(
+  [
+    'The row grouping have been moved to the new `@mui/x-data-grid-premium` package',
+    'Your application should continue to work with `@mui/x-data-grid-pro` but all the row-grouping related features will not be present',
```
Comment (m4theushw) [inline packages/grid/x-data-grid-pro/src/DataGridPro/useDataGridProProps.ts:33]:
```suggestion
    'MUI: The row grouping has been moved to the new `@mui/x-data-grid-premium` package.',
    'Your application should continue to work with `@mui/x-data-grid-pro` but all the row-grouping related features will not be present.',
```

**Review by m4theushw — COMMENTED** [review]
The changes to the docs will be done in another PR?

- The list of demos is empty in https://deploy-preview-4223--material-ui-x.netlify.app/api/data-grid/data-grid-premium/
- https://deploy-preview-4223--material-ui-x.netlify.app/components/data-grid/getting-started/#installation doesn't mention `@mui/x-data-grid-premium`

File: packages/grid/x-data-grid-premium/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+```sh
+// with npm
+npm install @mui/x-data-grid-premium
+
+// with yarn
+yarn add @mui/x-data-grid-premium
+```
+
+This component has three peer dependencies that you will need to install as well.
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply packages/grid/x-data-grid-premium/README.md:18]: I copied the content of the other packages but I agree that it is wrong
I fixed it for both pro / premium and community

**Comment by flaviendelangle** [comment]
@m4theushw @joserodolfofreitas 

> https://deploy-preview-4223--material-ui-x.netlify.app/components/data-grid/getting-started/#installation doesn't mention @mui/x-data-grid-premium

For this page, how would you organize it now that we have two commercial packages ?

**Comment by flaviendelangle** [comment]
> The list of demos is empty in https://deploy-preview-4223--material-ui-x.netlify.app/api/data-grid/data-grid-premium/

Fixed 
Related to https://github.com/mui/mui-x/issues/3135#issuecomment-1072407305, we will have to improve this logic

**Review by m4theushw — APPROVED** [review]
LGTM

**Pull request merged**